src/components/messageSpace/messageSpace.tsx

```diff
@@ -88,7 +88,65 @@ export default function MessageSpace(props: MessageSpaceProps) {
     return () => {
       targetDiv && intersectionObserver.unobserve(targetDiv)
     }
-  }, [areVideosLoaded, isScrolledToBottom, props.messages?.length])
+  }, [isScrolledToBottom])
+
+  function scrollDownIfNeeded() {
+    if (getVideoStatus()) {
+      const container = containerRef.current
+      setAreVideosLoaded(true)
+      if (container) {
+        // Use setTimeout to delay smooth scrolling so that it can scroll to bottom
+        setTimeout(() => {
+          hideScrollButton()
+          container.scrollTop = container.scrollHeight
+        }, 0)
+      }
+    } else {
+      setTimeout(scrollDownIfNeeded, 1)
+    }
+  }
+
+  useEffect(() => {
+    scrollDownIfNeeded()
+  }, [areVideosLoaded])
+
+  function scrollToLastMessage() {
+    if (getVideoStatus()) {
+      const container = containerRef.current
+      const childrenComponent = container && container.children
+      // scroll end div is also accounted as children component
+      const minimumChildrenLength = 2
+      const messageMargin = 32
```

Comment:
What does this message margin refer to? Would this still work properly if the margins got overwritten?

Reply:
I was trying to add space above the message, because without it, the sender name won't be shown. I couldn't figure out which part causes this. There's no margin/padding for messageSpace and messageCanvas. I tried to remove the gap between messages and it didn't change anything